Cannot start docker container: SIGILL exit code

**Describe the bug**
Running `sudo docker run --network=host --ulimit memlock=-1 docker.dragonflydb.io/dragonflydb/dragonfly` results in SIGILL abort.

**To Reproduce**
Steps to reproduce the behavior:
1. Run `sudo docker run --network=host --ulimit memlock=-1 docker.dragonflydb.io/dragonflydb/dragonfly`

**Expected behavior**
1. `sudo docker run --network=host --ulimit memlock=-1 docker.dragonflydb.io/dragonflydb/dragonfly` should not fail to run.
2. `docker ps` should show dragonfly.

**Environment (please complete the following information):**
 - OS: `Ubuntu 22.04.1 LTS`
 - Kernel: `Linux desktop 5.15.0-48-generic #54-Ubuntu SMP Fri Aug 26 13:26:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
 - Containerized?: Docker running on bare metal
 - Dragonfly Version: not sure

I tried compiling from HEAD (`ca77611` ). A few warnings were produced, but the resulting binary still crashes with SIGILL.

```
$ ninja dragonfly
... ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [16, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset 4 into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset 4 into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [5, 2147483626] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [5, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [6, 2147483625] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [6, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [7, 2147483624] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [7, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [8, 2147483623] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [8, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [9, 2147483622] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [9, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [10, 2147483621] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [10, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [11, 2147483620] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [11, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [12, 2147483619] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [12, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [13, 2147483618] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [13, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [14, 2147483617] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [14, 1073741823] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:228:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:227:15: note: at offset [15, 2147483616] into destination object ‘f’ of size 4
  227 |         float f = (float)luaL_checknumber(L, arg++);
      |               ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [16, 1073741823] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset 8 into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset 8 into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [9, 2147483622] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [9, 1073741823] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [10, 2147483621] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [10, 1073741823] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [11, 2147483620] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [11, 1073741823] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [12, 2147483619] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [12, 1073741823] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [13, 2147483618] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [13, 1073741823] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [14, 2147483617] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:194:14: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  194 |       b[i++] = b[size];
      |       ~~~~~~~^~~~~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [14, 1073741823] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
In function ‘correctbytes’,
    inlined from ‘correctbytes’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:189:13,
    inlined from ‘b_pack’ at /home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:234:9:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:195:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  195 |       b[size] = temp;
      |       ~~~~~~~~^~~~~~
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: In function ‘b_pack’:
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c:233:16: note: at offset [15, 2147483616] into destination object ‘d’ of size 8
  233 |         double d = luaL_checknumber(L, arg++);
      |                ^
/home/yonas/git/dragonfly/src/redis/lua/struct/lua_struct.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-undefined-inline’ may have been intended to silence earlier diagnostics
...

$ ./dragonfly --alsologtostderr
'./dragonfly --alsologtostderr' terminated by signal SIGILL (Illegal instruction)
```

--- a/src/server/CMakeLists.txt
+++ b/src/server/CMakeLists.txt
@@ -1,6 +1,12 @@
 add_executable(dragonfly dfly_main.cc)
 cxx_link(dragonfly base dragonfly_lib)
 
+if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND CMAKE_BUILD_TYPE STREQUAL "Release")
+  # Add core2 only to this file, thus avoiding instructions in this object file that
+  # can cause SIGILL.
+  set_source_files_properties(dfly_main.cc PROPERTIES COMPILE_FLAGS -march=core2)
+endif()
+
 add_library(dfly_transaction db_slice.cc engine_shard_set.cc blocking_controller.cc common.cc
             io_mgr.cc journal/journal.cc journal/journal_slice.cc table.cc
             tiered_storage.cc transaction.cc)

